File: Stage16/Python/Problem9012.py
import sys
input = sys.stdin.readline

def is_vps(s: str) -> bool:
    stack = []
    
    for ch in s:
        if ch == '(':
            stack.append(ch)
        else:
            if not stack:
                return False
            stack.pop()

    return not stack

def main():
    T = int(input())
    for _ in range(T):
        s = input().strip()
        print("YES" if is_vps(s) else "NO")

if __name__ == "__main__":
    main()

File: Stage16/Java/Problem9012.java
package Stage16.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Problem9012 {
    static boolean isVPS(String s) {
        Stack<Character> stack = new Stack<>();

        for (char c : s.toCharArray()) {
            if (c == '(') stack.push(c);
            else {
                if (stack.isEmpty()) return false;
                stack.pop();
            }
        }

        return stack.isEmpty();
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());
        StringBuilder sb = new StringBuilder();

        while (T-- > 0) {
            String str = br.readLine();
            sb.append(isVPS(str) ? "YES\n" : "NO\n");
        }

        System.out.print(sb);
        br.close();
    }
}
